internal/providers/vault/issue_257.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 Namespace Labs Inc; All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+
+package vault
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/vault-client-go"
+)
+
+// Temporary workaround until the fix is merged:
+// https://github.com/hashicorp/vault-client-go/pull/260
+func withIssue257Workaround() vault.ClientOption {
+	c := vault.DefaultConfiguration().HTTPClient
+	c.Transport = fix257{rt: c.Transport}
+	return vault.WithHTTPClient(c)
+}
+
+type fix257 struct {
+	rt http.RoundTripper
+}
+
+func (w fix257) RoundTrip(req *http.Request) (*http.Response, error) {
+	if !strings.HasSuffix(req.URL.Path, "/secret-id") {
+		return w.rt.RoundTrip(req)
+	}
+	res, err := w.rt.RoundTrip(req)
+	if err != nil {
+		return res, err
+	}
+	defer res.Body.Close()
```

Comment:
This is actually not correct: if we don't consume and replace the body, we shouldn't close it. It was in fact correct earlier when it wasn't closed after that long if statement :P

Reply:
Added a comment to make the intention clear here.